Client/Utils/FaviconFetcher.swift

```diff
@@ -56,6 +57,31 @@ open class FaviconFetcher: NSObject, XMLParserDelegate {
         }
         return nil
     }
+    
+    class func getCustomIcon(for url: URL) -> (url: URL, color: UIColor)? {
+        let dirName = NTPDownloader.faviconOverridesDirectory
+        
+        guard let folder = FileManager.default.getOrCreateFolder(name: dirName),
+            let baseDomain = url.baseDomain else { return nil }
+        
+        let backgroundFileName = baseDomain + NTPDownloader.faviconOverridesBackgroundSuffix
+        let backgroundFilePath = folder.appendingPathComponent(backgroundFileName)
+        
+        do {
+            let colorString = try String(contentsOf: backgroundFilePath)
+                .replacingOccurrences(of: "#", with: "")
```

Comment:
Move this code into the convenience init, since it doesn't do anything if `#` is absent, it shouldn't impact current usage at all.